File: app/src/main/java/com/soutvoid/gamesproject/domain/game/Game.java
package com.soutvoid.gamesproject.domain.game;

import com.soutvoid.gamesproject.domain.Image;

import java.io.Serializable;
import java.lang.reflect.Field;
import java.util.ArrayList;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;


@Data
@AllArgsConstructor
@EqualsAndHashCode
public class Game implements Serializable {

    private Integer id;
    private String name;
    private String slug;
    private String url;
    private Long createdAt;
    private Long updatedAt;
    private String summary;
    private String storyline;
    private Integer collection;
    private Long franchise;
    private Double rating;
    private Integer rating_count;
    private Integer hypes;
    private Double popularity;
    private Double aggregatedRating;
    private Long game;
    private ArrayList<Integer> developers = null;
    private ArrayList<Integer> publishers = null;
    private ArrayList<Long> gameEngines = null;
    private Integer category;
    private TimeToBeat timeToBeat;
    private ArrayList<Integer> playerPerspectives = null;
    private ArrayList<Integer> gameModes = null;
    private ArrayList<Integer> keywords = null;
    private ArrayList<Integer> themes = null;
    private ArrayList<Integer> genres = null;
    private Long firstReleaseDate;
    private Integer status;
    private ArrayList<ReleaseDate> releaseDates = null;
    private ArrayList<AlternativeName> alternativeNames = null;
    private ArrayList<Image> screenshots = null;
    private ArrayList<Video> videos = null;
    private Image cover;
    private Esrb esrb;
    private Pegi pegi;

    @Override
    public String toString() {
        String result = "";
        Field[] fields = this.getClass().getDeclaredFields();
        for (Field field :
                fields) {
            try {
                result += field.getName() + ": " + field.get(this).toString() + "\n";
            } catch (IllegalAccessException | NullPointerException e) {
                //ignore
            }
        }
        return result;
    }


}

File: app/src/main/java/com/soutvoid/gamesproject/domain/game/Game.kt
package com.soutvoid.gamesproject.domain.game

import com.soutvoid.gamesproject.domain.Image
import java.io.Serializable
import java.util.*

data class Game(
        val id: Int,
        val name: String,
        val slug: String?,
        val url: String?,
        val createdAt: Long?,
        val updatedAt: Long?,
        val summary: String?,
        val storyline: String?,
        val collection: Int?,
        val franchise: Long?,
        val rating: Double?,
        val rating_count: Int?,
        val hypes: Int?,
        val popularity: Double?,
        val aggregatedRating: Double?,
        val game: Long?,
        val developers: ArrayList<Int>?,
        val publishers: ArrayList<Int>?,
        val gameEngines: ArrayList<Long>?,
        val category: Int?,
        val timeToBeat: TimeToBeat?,
        val playerPerspectives: ArrayList<Int>?,
        val gameModes: ArrayList<Int>?,
        val keywords: ArrayList<Int>?,
        val themes: ArrayList<Int>?,
        val genres: ArrayList<Int>?,
        val firstReleaseDate: Long,
        val status: Int?,
        val releaseDates: ArrayList<ReleaseDate>?,
        val alternativeNames: ArrayList<AlternativeName>?,
        val screenshots: ArrayList<Image>?,
        val videos: ArrayList<Video>?,
        val cover: Image?,
        val esrb: Esrb?,
        val pegi: Pegi?
) : Serializable
